src/streamsync/core.py

```diff
@@ -477,30 +477,51 @@ def call_frontend_function(self, module_key: str, function_name: str, args: List
 
 class Component:
 
-    def __init__(self, id: str, type: str, content: Dict[str, str] = {}):
+    def __init__(self, id: str, type: str, content: Dict[str, str] = {}, flag: Optional[str] = None):
         self.id = id
         self.type = type
+        self.flag = flag
         self.content = content
         self.position: int = 0
         self.parentId: Optional[str] = None
         self.handlers: Optional[Dict[str, str]] = None
         self.visible: Optional[bool] = None
         self.binding: Optional[Dict] = None
 
+    @classmethod
+    def from_dict(cls, c_dict: Dict) -> 'Component':
```

Comment:
I'll try it. Hadn't had too much experience with Pydantic, but I know my way around. I'll keep you updated.

Reply:
Was pretty straightforward, as Component class is not widely used across the codebase (solid architecture!). And this clarifies the code in some places, so definitely was worth it. I'm slightly worried that it may lead to some issues with validation, but for now it seems all good.